pkg/proc/native/syscall_windows_arm64.go

```diff
@@ -0,0 +1,35 @@
+package native
+
+import (
+	"syscall"
+
+	"github.com/go-delve/delve/pkg/proc/winutil"
+)
+
+const (
+	_CONTEXT_ARM64               = 0x00400000
+	_CONTEXT_CONTROL             = (_CONTEXT_ARM64 | 0x1)
+	_CONTEXT_INTEGER             = (_CONTEXT_ARM64 | 0x2)
+	_CONTEXT_FLOATING_POINT      = (_CONTEXT_ARM64 | 0x4)
+	_CONTEXT_DEBUG_REGISTERS     = (_CONTEXT_ARM64 | 0x8)
+	_CONTEXT_ARM64_X18           = (_CONTEXT_ARM64 | 0x10)
+	_CONTEXT_FULL                = (_CONTEXT_CONTROL | _CONTEXT_INTEGER | _CONTEXT_FLOATING_POINT)
+	_CONTEXT_ALL                 = (_CONTEXT_CONTROL | _CONTEXT_INTEGER | _CONTEXT_FLOATING_POINT | _CONTEXT_DEBUG_REGISTERS | _CONTEXT_ARM64_X18)
+	_CONTEXT_EXCEPTION_ACTIVE    = 0x8000000
+	_CONTEXT_SERVICE_ACTIVE      = 0x10000000
+	_CONTEXT_EXCEPTION_REQUEST   = 0x40000000
+	_CONTEXT_EXCEPTION_REPORTING = 0x80000000
+)
+
+// zsyscall_windows.go, an autogenerated file, wants to refer to the context
+// structure as _CONTEXT, but we need to have it in pkg/proc/winutil.CONTEXT
+// because it's also used on non-windows operating systems.
+type _CONTEXT = winutil.ARM64CONTEXT
+
+func GetThreadContext(thread syscall.Handle, context winutil.CONTEXT) (err error) {
```

Comment:
We don't want to export this function. But also I'm not a fan of having winutil.CONTEXT be an interface, at least in pkg/proc/native. Because it doesn't actually have a dynamic type, inside pkg/proc/native it will always have the type specific to the architecture we are running on. The code in pkg/proc/native should use the concrete type, code that isn't architecture specific should still manipulate the concrete type using helper methods.
For example `setContextFlags(...)` which is implemented as `setContextFlags(*winutil.AMD64CONTEXT,uint32)` on amd64, and as `setContextFlags(*winutil.ARM64CONTEXT,uint32)` on arm64.

I see you are having problems with minidumps, I think the code there would be clearer and easier to write with concrete types also (but I don't know what the problem is so I could be wrong).

Reply:
Agree and done! Thanks for the examples, they helped me with the refactor.